### DISCUSSION TITLE
<CodeHighlightTabs /> expandable code threshold

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#6933] [Feature requests]
We want to set a threshold (height or lines) of when the code becomes "expandable" because currently it's hardcoded:
[(https://mantine.dev/x/code-highlight/#expandable-code)](https://mantine.dev/x/code-highlight/#expandable-code)
![Screenshot 2024-10-02 at 15 47 41](https://github.com/user-attachments/assets/8d9b7b70-19b1-4bdc-80ff-db39056eccfa)

**Comment by Kenzo-Wada** [comment]
Does the prop `maxCollapsedHeight` not seem to solve the problem?

**Reply by atumas-bananamilk** [reply]
Where did you find this? It's not in the docs.

**Reply by atumas-bananamilk** [reply]
`maxCollapsedHeight` is different, it just sets the height of collapsed item.
`maxCollapsedHeight` = 100:
![Screenshot 2024-12-11 at 19 53 00](https://github.com/user-attachments/assets/1424bb3f-bc31-4f28-9640-8161bb7f110b)

`maxCollapsedHeight` = 200:
![Screenshot 2024-12-11 at 19 53 07](https://github.com/user-attachments/assets/94bfdc05-a2b0-497a-8ad9-55178bb48181)

I need "show full" to show up only at some height threshold.

If height of this is 200px, I should see this:
![Screenshot 2024-12-11 at 19 54 30](https://github.com/user-attachments/assets/8bbd8e34-a212-4391-a36c-3117aadceb45)

If height of this is >200px, I should see this:
![Screenshot 2024-12-11 at 19 55 45](https://github.com/user-attachments/assets/ec4cae0f-f1eb-4ae8-b07f-6b2b12d6f056)

**Comment by atumas-bananamilk** [comment]
Current solution:
```
classNames={{
   codeWrapper: countLines(code) > lineThreshold ? 'min-h-[300px]' : '',
 }}
```
